src/components/split-button/split-button.component.js

```diff
@@ -17,108 +17,99 @@ const marginPropTypes = filterStyledSystemMarginProps(
-    this.isToggleButtonFocused = true;
-    this.showButtons();
-  };
-
-  showButtons = () => {
-    document.addEventListener(this.userInputType, this.handleClickOutside);
-    this.setState({
-      showAdditionalButtons: true,
-      minWidth:
-        CONTENT_WIDTH_RATIO *
-        this.splitButtonNode.current.getBoundingClientRect().width,
-    });
-
-    if (!this.listening) {
-      document.addEventListener("keydown", this.handleKeyDown);
-      this.listening = true;
-    }
-  };
-
-  hideButtons = () => {
-    if (this.isToggleButtonFocused) return;
+const SplitButton = ({
+  align = "left",
+  as = "secondary",
+  buttonType,
+  children,
+  disabled = false,
+  iconPosition = "before",
+  iconType,
+  onClick,
+  size = "medium",
+  subtext,
+  text,
+  ...rest
+}) => {
+  const isToggleButtonFocused = useRef(false);
+  const listening = useRef(false);
+  const buttonLabelId = useRef(guid());
+  const additionalButtons = useRef([]);
+  const splitButtonNode = useRef();
+  const buttonContainer = useRef();
+  const [showAdditionalButtons, setShowAdditionalButtons] = useState(false);
+  const [minWidth, setMinWidth] = useState(0);
+  const userInputType = useRef(
+    "ontouchstart" in document.documentElement ? "touchstart" : "click"
+  );
+
+  const hideButtons = useCallback(() => {
+    if (isToggleButtonFocused.current) return;
+
+    setShowAdditionalButtons(false);
+    listening.current = false;
+  }, []);
+
+  const handleClickOutside = useCallback(
+    ({ target }) => {
+      if (
+        !splitButtonNode.current.contains(target) &&
+        buttonContainer.current &&
+        !buttonContainer.current.contains(target)
+      ) {
+        hideButtons();
+      }
+    },
+    [hideButtons]
+  );
+
+  const handleKeyDown = useCallback(
+    (ev) => {
+      if (!listening.current) {
+        return;
+      }
 
-    this.setState({ showAdditionalButtons: false });
-    document.removeEventListener(this.userInputType, this.handleClickOutside);
+      const numOfChildren = children.length - 1;
+      const currentIndex = additionalButtons.current.findIndex(isActiveElement);
+      let nextIndex = -1;
 
-    if (this.listening) {
-      document.removeEventListener("keydown", this.handleKeyDown);
-      this.listening = false;
-    }
-  };
-
-  handleClickOutside = ({ target }) => {
-    if (
-      !this.splitButtonNode.current.contains(target) &&
-      this.buttonContainer.current &&
-      !this.buttonContainer.current.contains(target)
-    ) {
-      this.hideButtons();
-    }
-  };
+      if (Events.isUpKey(ev)) {
+        nextIndex = currentIndex > 0 ? currentIndex - 1 : numOfChildren;
+        ev.preventDefault();
+      }
+      if (Events.isDownKey(ev)) {
+        nextIndex = currentIndex < numOfChildren ? currentIndex + 1 : 0;
+        ev.preventDefault();
+      } else if (Events.isTabKey(ev)) {
+        // timeout enforces thet the "hideButtons" method will be run after browser focuses on the next element
```

Comment:
```suggestion
        // timeout enforces that the "hideButtons" method will be run after browser focuses on the next element
```

Reply:
wow - this typo was there for more than two years now! Fixed.